function.go

```diff
@@ -150,23 +169,34 @@ func NewFunctionWith(defaults Function) Function {
 // Functions from earlier versions are brought up to current using migrations.
 // Migrations are run prior to validators such that validation can omit
 // concerning itself with backwards compatibility.  Migrators must therefore
-// selectively consider the minimal validation necesssary to ehable migration.
+// selectively consider the minimal validation necessary to enable migration.
 func NewFunction(path string) (f Function, err error) {
-	f.Root = path // path is not persisted, as this is the purvew of the FS itself
+	f.Root = path // path is not persisted, as this is the purview of the FS itself
 	var filename = filepath.Join(path, FunctionFile)
 	if _, err = os.Stat(filename); err != nil {
 		return
 	}
-	bb, err := ioutil.ReadFile(filename)
+	bb, err := os.ReadFile(filename)
 	if err != nil {
 		return
 	}
-	if err = yaml.Unmarshal(bb, &f); err != nil {
-		err = formatUnmarshalError(err) // human-friendly unmarshalling errors
-		return
+	var functionMarshallingError error
```

Comment:
@lkingland @lance help is appreciated here.. I do not like this error handling, but there was nothing in place to cover this use cases. For now I think this is ok to report back the errors (both marshalling and migration) but it can be improved in the future. If you know about other places where we are dealing with similar errors in a different way please point me in that direction or we can create another issue to improve this code later on.

Reply:
Off the top of my head, I cannot think of a place in the code where we are handling yaml marshaling differently. I do agree that this could be improved. A new issue would be good for that, imo. For now this works.

Reply:
Agreed with @lance we can improve this in a separate PR; with this working for now.  I might just upgrade to typed errors (see other comment for details)